### PR TITLE
feat: added helm charts + dev env

### LINKED ISSUE #1: Setup Project

### PR DESCRIPTION
**Pull request by Al-Pragliola**
- added helm charts
- added dev env with Tilt

closes #1

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+steps:
+  - name: prepare
+    image: quay.io/sighup/golang:1.20.4
+    depends_on:
+      - clone
+    pull: always
+    commands:
+      - mkdir -p .go/cache .go/modcache .go/tmp
+      - go mod download
+  - name: unit-test
```
Comment (omissis) [inline .drone.yml:141]: nitpick
```suggestion
      - go mod download

  - name: unit-test
```

File: .drone.yml [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+      - docker buildx create --name sighup-builder
+      - docker buildx use sighup-builder
+      - docker buildx inspect --bootstrap
+      - docker buildx build --platform linux/amd64,linux/arm64,linux/arm/v7 \
+        -t "quay.io/sighup/fury-ui-backend:${DRONE_TAG}" --push .
+      - docker buildx build --platform linux/amd64,linux/arm64,linux/arm/v7 \
+        -t "quay.io/sighup/fury-ui-frontend:${DRONE_TAG}" --push ./web-client
+
+  - name: release
+    image: golang:1.20
```
Comment (omissis) [inline .drone.yml:205]: maybe we could use `quay.io/sighup/golang:1.20.4` if it included `goreleaser` as well?

File: .gitignore [hunk 3]
```diff
@@ -4,5 +4,10 @@ test.*
 .DS_Store
 tmp
 .envrc
+.kubeconfig
```
Comment (omissis) [inline .gitignore:7]: not sure why we started using `.kubeconfig` instead of `kubeconfig`?

File: development/down.sh
Comment (omissis) [inline development/down.sh:1]: add a `docker stop fury-ui-kind-registry`, either at line 11, or add an else (if force == 1 then remove else stop)

File: Tiltfile [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    values='development/helm/values.yaml',
+#    set=['config.controlPlaneAddress=' + os.getenv("CONTROL_PLANE_ADDRESS")]
+    set=[]
+  )
+)
+
+k8s_resource(
+  workload="fury-ui-backend",
+  links=[
+    link("https://fury-ui.dev/", "fury-ui-backend"),
```
Comment (omissis) [inline Tiltfile:49]: perhaps it should point at the api endpoint, or to a swagger ui?
```suggestion
    link("https://fury-ui.dev/api/v1", "fury-ui-backend"),
```

File: helm_chart/Chart.yaml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.0.1
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+appVersion: 0.0.1
```
Comment (omissis) [inline helm_chart/Chart.yaml:51]: If I remember correctly, semver requires to start at `v0.1.0`

File: helm_chart/charts/fury-ui-backend/Chart.yaml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.0.1
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+appVersion: 0.0.1
```
Comment (omissis) [inline helm_chart/charts/fury-ui-backend/Chart.yaml:34]: If I remember correctly, semver requires to start at `v0.1.0`

File: helm_chart/charts/fury-ui-backend/values.yaml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+# Backend
+
+image:
+  repository: quay.io/sighup/fury-ui-backend
+  pullPolicy: IfNotPresent
+  # Overrides the image tag whose default is the chart appVersion.
+  tag: "0.0.1"
```
Comment (omissis) [inline helm_chart/charts/fury-ui-backend/values.yaml:15]:
```suggestion
  tag: "0.1.0"
```

File: helm_chart/charts/fury-ui-frontend/values.yaml [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+# Frontend
+
+image:
+  repository: quay.io/sighup/fury-ui-frontend
+  pullPolicy: IfNotPresent
+  # Overrides the image tag whose default is the chart appVersion.
+  tag: "0.0.1"
```
Comment (omissis) [inline helm_chart/charts/fury-ui-frontend/values.yaml:15]:
```suggestion
  tag: "0.1.0"
```

File: helm_chart/charts/fury-ui-frontend/values.yaml [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+nodeSelector: {}
+
+tolerations: []
+
+affinity: {}
+
+# Frontend settings file:
+settings:
+  # The URL of the backend API.
+  apiUrl: "http://example.com/api/v1"
```
Comment (omissis) [inline helm_chart/charts/fury-ui-frontend/values.yaml:69]:
```suggestion
  apiUrl: "http://fury.dev/api/v1"
```

File: web-client/src/logo.svg
Comment (omissis) [inline web-client/src/logo.svg:1]: can you make the logo bigger? 😬 `</troll>`

File: development/up.sh
Comment (omissis) [inline development/up.sh:1]: `mv development/up.sh scripts/dev-up.sh`

File: development/utils.sh
Comment (omissis) [inline development/utils.sh:1]: `mv development/utils.sh scripts/dev-utils.sh`

Comment (omissis) [reply development/utils.sh:1]: remember to update the references to utils.sh

File: development/down.sh
Comment (omissis) [inline development/down.sh:1]: `mv development/down.sh scripts/dev-down.sh`

Comment (omissis) [reply development/down.sh:1]: development isn't a go standard layout folder, so I would skip it

File: development/helm/ingress-nginx.yaml
Comment (omissis) [inline development/helm/ingress-nginx.yaml:1]: `mv development/helm/ingress-nginx.yaml scripts/helm-values/ingress-nginx.yaml`

File: development/helm/values.yaml
Comment (omissis) [inline development/helm/values.yaml:1]: `mv development/helm/values.yaml development/helm-values/fury-ui.yaml`

File: development/manifests/ctlptl-registry.yml
Comment (omissis) [inline development/manifests/ctlptl-registry.yml:1]: `mv development/manifests/ctlptl-registry.yml scripts/kubernetes-manifests/ctlptl-registry.yml`

File: development/manifests/kind-config.yml
Comment (omissis) [inline development/manifests/kind-config.yml:1]: `mv development/manifests/kind-config.yml scripts/kubernetes-manifests/kind-config.yml`

File: development/tiltfiles/ingress-nginx.tiltfile
Comment (omissis) [inline development/tiltfiles/ingress-nginx.tiltfile:1]: `mv development/tiltfiles/ingress-nginx.tiltfile configs/tiltfiles/ingress-nginx.tiltfile`

File: development/tiltfiles/setup.tiltfile
Comment (omissis) [inline development/tiltfiles/setup.tiltfile:1]: `mv development/tiltfiles/setup.tiltfile configs/tiltfiles/setup.tiltfile`

**Review by omissis — CHANGES_REQUESTED** [review]

File: scripts/dev-down.sh [hunk 10]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
```
Comment (omissis) [inline scripts/dev-down.sh:1]:
```suggestion
#!/bin/sh -x
```

File: scripts/dev-up.sh [hunk 11]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
```
Comment (omissis) [inline scripts/dev-up.sh:1]:
```suggestion
#!/bin/sh -x
```

**Review by omissis — CHANGES_REQUESTED** [review]

File: helm_chart/values.yaml [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+#   correct namespace.
+# - If you're using cert-manager.io to provision your certificates, just uncomment "cert-manager.io/cluster-issuer" and
+#   type the name of your cluster-issuer. Please also uncomment the "tls" part with default values.
+ingress:
+  enabled: false
+  annotations:
+  # kubernetes.io/ingress.class: nginx
+  # cert-manager.io/cluster-issuer: my-cluster-issuer
+  hosts:
+    - host: fury-ui.domain.com
```
[the remark this answers is not in the document]
Comment (Al-Pragliola) [reply helm_chart/values.yaml:28]: This file is for public usage, not for tilt, isn't it more helpful to keep a generic domain?

**Review by omissis — APPROVED** [review]

**Pull request merged**